src/components/HotspotEditorModal/HotspotTextStyleTab/HotspotTextStyleTab.jsx

```diff
@@ -0,0 +1,256 @@
+  borderColors: undefined,
+};
+
+/* this component is only used internally where props are defined and set. */
+const HotspotTextStyleTab = ({
+  className,
+  fontColors,
+  backgroundColors,
+  borderColors,
+  formValues,
+  light,
+  i18n,
+  onChange,
+}) => {
+  const {
+    boldLabel,
+    italicLabel,
+    underlineLabel,
+    fontLabel,
+    fontSizeLabel,
+    fontSizeInvalid,
+    backgroundLabel,
+    fillOpacityLabel,
+    fillOpacityInvalid,
+    borderLabel,
+    borderWidthLabel,
+    borderWidthInvalid,
+  } = merge({}, defaultProps.i18n, i18n);
+
+  const fontSizeRef = useRef(null);
+  const fillOpacityRef = useRef(null);
+  const borderWidthRef = useRef(null);
+
+  const { bold, italic, underline, font, background, border } = merge(
+    {},
+    defaultProps.formValues,
+    formValues
+  );
+
+  return (
+    <div className={className}>
+      <div className={`${iotPrefix}--hotspot-text-style-tab__text-style`}>
+        <IconSwitch
+          onClick={() => onChange({ bold: !bold })}
+          data-testid="hotspot-bold"
+          selected={bold}
+          text={boldLabel}
+          renderIcon={TextBold}
+          index={0}
+          light={light}
+        />
+        <IconSwitch
+          name="italic"
+          onClick={() => onChange({ italic: !italic })}
+          data-testid="hotspot-italic"
+          selected={italic}
+          text={italicLabel}
+          renderIcon={TextItalic}
+          index={1}
+          light={light}
+        />
+        <IconSwitch
+          name="underline"
+          onClick={() => onChange({ underline: !underline })}
+          data-testid="hotspot-underline"
+          selected={underline}
+          text={underlineLabel}
+          renderIcon={TextUnderline}
+          index={2}
+          light={light}
+        />
+      </div>
+
+      <div className={`${iotPrefix}--hotspot-text-style-tab__row`}>
+        <ColorDropdown
+          id={`${iotPrefix}--hotspot-text-style-tab__font-color`}
+          titleText={fontLabel}
+          light={light}
+          selectedColor={font.color ?? fontColors?.[0]}
+          colors={fontColors}
+          onChange={(selected) => {
+            onChange({ font: selected });
+          }}
+        />
+
+        <NumberInput
+          id={`${iotPrefix}--hotspot-text-style-tab__font-size`}
+          min={0}
+          max={50}
+          value={font.size}
+          step={1}
+          light={light}
+          label={fontSizeLabel}
+          invalidText={fontSizeInvalid}
+          ref={fontSizeRef}
+          onChange={() => {
+            onChange({ font: { size: fontSizeRef.current.value } });
+          }}
+        />
+      </div>
+
+      <div className={`${iotPrefix}--hotspot-text-style-tab__row`}>
+        <ColorDropdown
+          id={`${iotPrefix}--hotspot-text-style-tab__background-color`}
+          titleText={backgroundLabel}
+          light={light}
+          selectedColor={background.color ?? backgroundColors?.[0]}
```

Comment:
Suggestion, skip the fallback `backgroundColors?.[0]` and just use `backgroundColor` which might be undefined which is okay.